mqtt/mqtt-broker/src/transport.rs

```diff
@@ -79,10 +41,17 @@ impl Transport {
         Ok(Transport::Tcp(tcp))
     }
 
-    async fn new_tls<A>(addr: A, identity: Identity) -> Result<Self, InitializeBrokerError>
+    pub async fn new_tls<A>(addr: A, cert_path: PathBuf) -> Result<Self, InitializeBrokerError>
```

Comment:
nit: What is the benefit of writing the cert to disk? Can we pass `Vec<u8>` instead of `PathBuf`? It can help eliminate storing the cert as a file in edgehub container.

Reply:
I thought about that, but in generic broker world it is hard to describe such a setting. It's not like we download cert from some location, we make very specific HTTP POST request to edgelet. I'm open to any suggestions :)

Reply:
>but in generic broker world it is hard to describe such a setting.

I'm not sure I understand that. 

What I suggested is just to change to  `fn new_tls<A>(addr: A, cert: Vec<u8>) -> Result<Self, InitializeBrokerError>`, and `fn dowload_server_certificate() -> Result<Vec<u8>>`

Then in `edgehub.rs`
```rust
    if let Some(tls) = config.transports().tls() {
        let cert = download_server_certificate().await?;
        server.tls(tls.addr(), cert, authenticator.clone())?;
    }
```
And in `generic.rs`
```rust
    if let Some(tls) = config.transports().tls() {
        let authenticator = authenticate_fn_ok(|_| Some(AuthId::Anonymous));
        let cert = fs::read(tls.cert_path())
            .map_err(|e| InitializeBrokerError::LoadIdentity(cert_path.to_path_buf(), e))?;
        server.tls(tls.addr(), cert, authenticator)?;
    }
```
Pros - you don't have to store edgehub server cert on disk.
does that make sense?

Reply:
this part is clear. what to do with settings? now tls section always require a path to identity cert.
for generic broker it is should stay the same : address + cert_path.
for edgehub we can omit that. but it means we need a different setting object for tls.